lessons/beginners/str/index.md

```diff
@@ -1,353 +1,317 @@
-  ╰───┴───┴───┴───┴───┴───┴───┴───╯
+```pycon
+>>> print("Vtom vnuk křik': " + '"Hleď!"')
+Vtom vnuk křik': "Hleď!"
 ```
 
+Ale lepší způsob je použít speciální zápis se *zpětným lomítkem*.
+Kdykoli se v řetězci objeví sekvence `\'` nebo `\"`, Python dá do řetězce danou
+uvozovku.
 
-A když už jsme u divných čísel,
-co se asi stane, když budu chtít vybírat písmena
-pomocí záporných čísel?
-
-{% filter solution %}
-```python
-print('Čokoláda'[-1])  # → a
-print('Čokoláda'[-2])  # → d
-print('Čokoláda'[-3])  # → á
-print('Čokoláda'[-4])  # → l
+```pycon
+>>> print("Vtom vnuk křik': \"Hleď!\"")
+Vtom vnuk křik': "Hleď!"
+>>> print('"Jen ho nech," řek\' děd. "Kdo zná líp kraj?"')
+"Jen ho nech," řek' děd. "Kdo zná líp kraj?"
 ```
 
-Záporná čísla vybírají písmenka od konce.
+Ve výsledném řetězci pak ovšem žádné zpětné lomítko *není*.
+Sekvence `\'` je jen způsob, jak v Pythonu zadat `'` – jediný znak.
+Tomu je celkem důležité porozumět.
+Zkus si, jestli zvládneš předpovědět výsledek těchto výrazů:
 
-```plain
-   [0] [1] [2] [3] [4] [5] [6] [7]
-   [-8][-7][-6][-5][-4][-3][-2][-1]
-  ╭───┬───┬───┬───┬───┬───┬───┬───╮
-  │ Č │ o │ k │ o │ l │ á │ d │ a │
-  ╰───┴───┴───┴───┴───┴───┴───┴───╯
+```pycon
+>>> print(".\".")
+>>> len(".\".")
+>>> ".\"."
 ```
-{% endfilter %}
-
-Řetězce umí i jiné triky.
-Třeba můžeš zjistit, jak je řetězec dlouhý
-nebo jestli v sobě obsahuje daný menší řetězec.
-
-<table class="table">
-    <tr>
-        <th>Zápis</th>
-        <th>Popis</th>
-        <th>Příklad</th>
-    </tr>
-    <tr>
-        <td><code>len(r)</code></td>
-        <td>Délka řetězce</td>
-        <td><code>len('čokoláda')</code></td>
-    </tr>
-    <tr>
-        <td><code>x&nbsp;in&nbsp;r</code></td>
-        <td>True pokud je řetězec <code>x</code> obsažen v <code>r</code></td>
-        <td><code>'oko' in 'čokoláda'</code></td>
-    </tr>
-    <tr>
-        <td><code>x&nbsp;not&nbsp;in&nbsp;r</code></td>
-        <td>Opak <code>x in r</code></td>
-        <td><code>'dub' not in 'čokoláda</code></td>
-    </tr>
-</table>
-
-Řetězce vždy berou v potaz velikost písmen,
-takže např. `'ČOKO' in 'čokoláda'` je `False`.
-Kdybys chtěl{{a}} porovnávat bez ohledu na velikost písmen,
-musel{{a}} bys oba řetězce převést třeba na malá písmena
-a pak je porovnat.
-
-A jak se převádí na malá písmena?
-K tomu budeme potřebovat další novou vlastnost Pythonu: metody.
-
-## Metody
-
-*Metoda* (angl. *method*) je jako funkce – něco, co se dá zavolat.
-Na rozdíl od funkce je svázaná s nějakým *objektem* (hodnotou).
-Volá se tak, že se za objekt napíše tečka,
-za ní jméno metody a za to celé se, jako u funkcí, připojí závorky
-s případnými argumenty.
-
-Řetězcové metody `upper()` a `lower()`
-převádí text na velká, respektive malá písmena.
 
-```python
-retezec = 'Ahoj'
-print(retezec.upper())
-print(retezec.lower())
-print(retezec)
+{% filter solution %}
+```pycon
+>>> print(".\".")
+.".
+>>> len(".\".")
+3
+>>> ".\"."
+'.".'
 ```
+{% endfilter %}
```

Comment:
This can be an interesting part for experimenting, but IMHO should be all hidden by default and the box with this part would display when clicked on: ```Zkus si, jestli zvládneš předpovědět výsledek těchto výrazů```

Reply:
I don't think this should be hidden. People struggle with the fact that `\"` denotes a single character. The teacher could show these as an example, or skip this and use some other way to explain it, but IMO it is necessary to show the fact again, from a different angle.